relation/index.js

```diff
@@ -108,7 +111,39 @@ module.exports = yeoman.generators.Base.extend({
           }
           return m;
         },
-        validate: validateName
+        validate: function(value) {
+
+            var done = this.async();
+
+            //validfn returns false if relationship 
+            //has the same name with properties
+            var validfn = function(value, cb) {
+
+              //set flag isValid
+              var isValid = true;
+           
+              modelDef.properties(function(err, list) {
+                if(err) return done(err);
+                else {
+                  //check each property name in the list
+                  async.each(list, function(property, callback) {
+                    if(property.name === value) isValid = false;
+                    callback();
+                  }, function(err) {
+                    if(err) return done(err);
+                    cb(null, isValid);
+                  });
+                }
+              });
+            };
+            
+            //return the result
+            validfn(value, function(err, result){
+              if(err) return done(err);
+              else done(result && validateName(value));
```

Comment:
IMO it's better to call `validateName` first. The operation is faster than checking naming clashes with properties, therefore the user will receive an error sooner. It will also simplify your code, since you can pass `done` directly as `validFn` callback.

```js
validFn(value, done);
```

BTW shouldn't the last line L143 in your code be `done(null, result && validateName(value))`?